src/runtime/src/core/map.rs

```diff
@@ -136,7 +139,25 @@ pub fn make_module() -> ValueMap {
 
     result.add_fn("sort", |vm, args| match vm.get_args(args) {
         [Map(m)] => {
-            m.contents_mut().data.sort_keys();
+            let m = m.clone();
+            let vm = vm.child_vm();
+
+            let mut error = None;
+
+            m.contents_mut()
+                .data
+                .sort_by(|key_a, _, key_b, _| match cmp(vm, key_a, key_b) {
```

Comment:
`quick_sort` breaks the loop on an error, while we can't do this in the `sort_by` approach and should go through the whole loop, even if an error occurred.

Reply:
I see.. I'm leaning towards favouring `sort_by` because:

- quicksort has a worst case of O(n^2) and is unstable vs. O(n log n) + stable for timsort, so we lose stability here. We could find another algorithm that maintains stability and has better worst case performance, but...
- I think it's preferable to optimize the performance of the success case over the error case, so it seems a more straight forward option to just use `sort_by`..

What do you think?

Reply:
Ah, I should've check what's behind sorting in Rust. Yeah, `sort_by` + early return you proposed below seems to be more performant.